### PR TITLE
Added base/mapping quality filters for HetPulldown

### PR DESCRIPTION
**Pull request by samuelklee**
Also removed hetAlleleFraction parameter as it was a holdover from previous versions of HetPulldown where allelic skew was given as input.

### CODE REVIEW

**Comment by samuelklee** [comment]
Evaluation is not quite complete, but it looks like results slightly improve or otherwise do not change drastically.  In any case, it seems that some sort of filtering should be done in this step, so I am of the opinion that we should merge even if concordance with HAPSEG decreases.

This was prompted by #355 but does not completely resolve that issue.  Looks like two hets are consistently pulled down on Y across a few normals...will continue investigating.

**Comment by LeeTL1220** [comment]
Despite the fact that we still see hets on Y, I am assuming that I should still review this PR.

File: src/main/java/org/broadinstitute/hellbender/tools/exome/GetHetCoverage.java [hunk 1]
```diff
@@ -72,14 +69,6 @@
     protected File tumorHetOutputFile;
 
     @Argument(
-            doc = "Heterozygous allele fraction.",
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/GetHetCoverage.java]: Does this mean that we need to change the workflow scripts?

File: src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java [hunk 2]
```diff
@@ -31,8 +31,12 @@
     private final File refFile;
     private final IntervalList snpIntervals;
 
+    private static final int NUMBER_OF_LOG_UPDATES = 20;
+    private static final double HET_ALLELE_FRACTION = 0.5;
+
     /** Set quality and read-depth thresholds for pulldown, interval threshold for indexing for SamLocusIterator. */
-    private static final int MIN_QUALITY = 0;
+    private static final int MIN_BASE_QUALITY = 20;
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java]: Are you sure that you want these hardcoded and not an advanced parameter with a default value?

File: src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java [hunk 3]
```diff
@@ -31,8 +31,12 @@
     private final File refFile;
     private final IntervalList snpIntervals;
 
+    private static final int NUMBER_OF_LOG_UPDATES = 20;
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java]: What is a log update?  Can you add a comment?

**Comment by LeeTL1220** [comment]
Minor comments... Back to you @samuelklee

File: src/main/java/org/broadinstitute/hellbender/tools/exome/GetHetCoverage.java (hunk 1 above)
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/GetHetCoverage.java]: yup, should be a minor change.  will issue a PR for it.

File: src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java (hunk 3 above)
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java (hunk 2 above)
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java]: done, exposed as minMQ and minBQ

**Comment by samuelklee** [comment]
Thanks, @LeeTL1220, I've addressed the PR comments.  Also, evaluation finished on the PRAD samples and it looks like MAE and RMSE either slightly improve or don't change much across various cuts of the samples.  So even though we lose ~5% of the old hets I'm comfortable with this change.

**Pull request merged**